torch/_inductor/codegen/triton.py

```diff
@@ -1488,7 +1532,11 @@ def load(self, name: str, index: sympy.Expr):
         # "other" below is a workaround for https://github.com/openai/triton/issues/737
         # for bool, even though it's likely subject to the same bug, setting `other` leads
         # to LLVM errors so we are skipping it for now
-        if ("tmp" in mask or "rmask" in mask) and V.graph.get_dtype(name) != torch.bool:
+        if (
+            (has_tmpmask or has_rindex)
+            and V.graph.get_dtype(name) != torch.bool
+            and indexing.has_mask()
```

Comment:
Does `has_tmpmask or has_rindex` implies that `indexing.has_mask()` is true?

Reply:
Oh, there is corner case where there is a rindex but the rmask is filtered? Realized when reading the discussion below.